app/src/main/java/berlin/mfn/naturblick/ui/fieldbook/fieldbook/FieldbookViewModel.kt

```diff
@@ -31,12 +31,52 @@ import kotlinx.coroutines.launch
 import java.util.UUID
 
 class FieldbookViewModel(
-    occurenceId: UUID?,
     application: Application,
     private val savedStateHandle: SavedStateHandle
 ) : AndroidViewModel(application) {
     private val operationDao = ObservationDb.getDb(application).operationDao()
     private val speciesDao = StrapiDb.getDb(application).speciesDao()
+
+    private suspend fun toFieldbookObservation(observation: Observation): FieldbookObservation {
+        return if (observation.newSpeciesId != null)
```

Comment:
maybe you should use the if / else only for a species variable, because all other parameters are the same